docs/describing_whats_in_classes.md

```diff
@@ -0,0 +1,36 @@
+# Expected methods and attributes of classes
+
+Within each technology class, ODIES expects certain methods to be defined by the developer.
+This doc page briefly discusses those methods and their impact within ODIES.
+
+## Performance model
+
+Each technology class must define `get_performance_model()`, which returns an OpenMDAO system.
+This system contains the physics model for the technology.
+For a converter, this computes the outputted resources based on the inputted resources.
+The input and output resources for the performance model are all arrays with length 8760 for hourly timestepped yearly data.
+
+For example, for an electrolyzer, the performance model is an OpenMDAO system whose inputs include electricity and whose outputs include hydrogen produced.
+
+## Cost model
+
+Each technology class must also define `get_cost_model()`, which returns an OpenMDAO system containing the cost model.
+Specifically, this system must output `CapEx` and `OpEx` values for the technology.
+These values are later used in financial modeled and cost breakdowns.
+
+## Financial model (optional)
+
+Each technology class can define `get_financial_model()`, which returns an OpenMDAO system containing the financial model.
+This would override any plant level financial model, and is useful for technologies that have unique financial considerations.
+This will also be more relevant as we develop non-single-owner capabilities.
+
+## Control model (optional, unused)
```

Comment:
A general comment is to watch out for defining undefined functionality that can come back to haunt us.

Reply:
Definitely, could be close to gold plating or worse!